### PR TITLE
Feature/1.11 efficient dman 2

### PR DESCRIPTION
**Pull request by xavierh**
Sorry I forgot to commit that change
# Changes proposed in this pull request:
- Citizen hut need food

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java [hunk 1]
```diff
@@ -116,6 +116,12 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
         }
     }
 
+    public boolean needsAnything()
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1161)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 36 issues
* ![MAJOR][MAJOR] 2 major
* ![MINOR][MINOR] 34 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingHome.java#L191](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java#L191): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingHome.java#L88](https://github.com/Minecolonies/minecolonies/blob/b117e465c4876bbe5f416e4fd879d7de0093fcbf/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java#L88): Replace this lambda with method reference 'Objects::nonNull'. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1612)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 35 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 34 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [BuildingHome.java#L192](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java#L192): Assign this magic number 5 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MINOR][MINOR] [blockout](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/blockout): Add a 'package-info.java' file to document the 'blockout' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [controls](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/blockout/controls): Add a 'package-info.java' file to document the 'controls' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [views](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/blockout/views): Add a 'package-info.java' file to document the 'views' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [BuildingHome.java#L88](https://github.com/Minecolonies/minecolonies/blob/95e0f6e413b20fc5e4bb71fe35049066d71391b2/src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java#L88): Replace this lambda with method reference 'Objects::nonNull'. (sonar.java.source not set. Assuming 8 or greater.) [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1612)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by marvin-bitterlich** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1043/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by Raycoms** [comment]
:shipit: 

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1043/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Pull request merged**